apm/README.md

```diff
@@ -0,0 +1,450 @@
+
+# muta monitor 部署
+
+## 工程目录结构
+```
+muta-monitor
+|
+|___ agent
+|    |___ docker-compose.yml
+|    |___ config
+|          |___ promtail
+|               |___ promtail-config.yaml
+|
+|___ monitor
+     |___ docker-compose.yml
+     |___ config
+          |___ grafana
+          |    |___ dashboards
+          |    |    |___ muta-benchmark.json
+          |    |    |___ muta-node.json
+          |    |
+          |    |___ provisioning
+          |          |___ dashboards
+          |          |    |___ dashboards.yaml
+          |          |
+          |          |___ datasources
+          |               |___ datasources.yaml                            
+          |___ loki
+          |    |___ loki-local-config.yaml
+          |
+          |___ promethues
+               |___ prometheus.yml
+
+
+```
+
+其中 agent 主要跟随 muta 部署，负责采集信息
+monitor 需要一台机器部署，主要运行目前的监控服务
+
+
+## Agent 详解
+主要 agent 如下:
+
+| agent | 功能 |
+| --- | --- |
+| aode-exporter | 采集机器信息(cpu, 内存等) |
+| jaeger-agent | tracing |
+| promtail-agent | 采集日志 |
+
+### agent 目录
+```
+|___ agent
+     |___ docker-compose.yml
+     |___ config
+           |___ promtail
+                |___ promtail-config.yaml
+ 
+```
+
+
+### docker-compose.yml
+```yml
+version: '3'
+services:
+  node-exporter:
+    image: quay.io/prometheus/node-exporter:v0.18.1
+    container_name: muta_node_exporter
+    command:
+      - '--path.rootfs=/host'
+      - '--collector.tcpstat'
+    restart: on-failure
+    network_mode: 'host'
+    pid: 'host'
+    volumes:
+      - /:/host:ro,rslave
+
+  jaeger-agent:
+    image: jaegertracing/jaeger-agent:1.18.1
+    container_name: muta_jaeger_agent
+    command:
+        # 这里将 jaeger-collector_ip 改成 jaeger-collector 部署节点的 ip 
```

Comment:
The docker-compose support using `.env` file keeping environment variables, so we can transfer some config to it.